### ISSUE TITLE
High CPU usage (200%) when running Nuxt dev server in Docker container

### ISSUE THREAD

**Issue by martinszeltins** [nuxt/nuxt#22910]
### Environment

My host system:

````
- Operating System: Linux (Ubuntu 20.04 LTS)
- Computer:         Lenovo Legion 5 (32GB RAM, i5 10th gen)
- Node Version:     v18.12.1
- Nuxt Version:     3.7.0
- CLI Version:      3.7.2
- Nitro Version:    2.6.2
- Package Manager:  pnpm@8.1.0
- Builder:          -
- User Config:      devtools, vite
- Runtime Modules:  -
- Build Modules:    -
````

### Reproduction

https://github.com/martinszeltins/nuxt-docker-cpu

Steps to reproduce the issue:

```bash
# Clone the repo
git clone https://github.com/martinszeltins/nuxt-docker-cpu.git
cd nuxt-docker-cpu

# Run the Docker container
docker-compose up -d
docker-compose logs --follow

# Wait for it to install dependencies and build the
# app (you will see it in the logs, it happens automatically)

# Run the Nuxt dev server
docker exec -it nuxt-docker-client pnpm run dev
```

### Describe the bug

I have tried to keep the reproduction as minimal as possible, it is a fresh Nuxt app put inside a Docker container, nothing more. The issue that I am seeing is that the CPU usage is very high, fluctuating between 100% and 200%. This makes my computer's fans spin up and it stays this way even when I am not doing anything. The app just stays open and does nothing and the CPU spins like crazy. To me it seems like the CPU should be close to 0% because nothing should be happening. _When running a plain Vue / Vite app dev server in a container I did not notice this issue, only with Nuxt._

This happens only when running the dev sever _**in Docker**_, not locally. Maybe Nuxt has a problem with Linux or being run from a Docker container, I'm not sure. What is Nuxt dev server doing all the time in the background that the CPU usage is so high? 🤔  As soon as I stop the dev server, the CPU usage goes back to normal.

I did not know how to investigate this issue further because I don't know the inner workings of Nuxt but it seems like some kind of resource leak is happening here. 🤷🏻‍♂️ 

Here is my CPU usage **BEFORE** I start the dev server:
![image](https://github.com/nuxt/nuxt/assets/34019878/35fcaf80-519f-443e-a2fa-347b2ef8c2c6)

And here is my CPU usage **AFTER** I have started the dev server:
![image](https://github.com/nuxt/nuxt/assets/34019878/a7bb0110-8871-47a5-90bf-d6920be23ad9)

**Comment by peteromano** [comment]
FYI, I exclusively use docker, it's lightning fast for me, including using nuxt layers.

**Comment by martinszeltins** [comment]
> FYI, I exclusively use docker, it's lightning fast for me, including using nuxt layers.
> 
> Using node 18 alpine

Can you please share your Docker setup? Are you running the dev server in Docker container?

**Comment by CodeDredd** [comment]
Are you running WSL2?

**Comment by martinszeltins** [comment]
> Are you running WSL2?

No, as you can see from my environment I'm running Ubuntu 20.04 Linux.

**Comment by CodeDredd** [comment]
Howv is it if you run `pnpm dev --host` ?

**Comment by peteromano** [comment]
@martinszeltins I'll update with my details in a bit

**Comment by Sandros94** [comment]
> How is it if you run `pnpm dev --host` ?

On my side I need to use this to have similar performance as of 3.6.5 (both cpu load and full-page loading times). Regardless of Nuxt running inside a container or not.

**Comment by martinszeltins** [comment]
> How is it if you run `pnpm dev --host` ?

Please check my reproduction, as you can see in my `package.json` - I am running the dev server with --host because that is the only way to run it inside a container.

**Comment by martinszeltins** [comment]
> @martinszeltins
> 
> Windows 36gb RAM WSL 2 Ubuntu 18.04 Docker Desktop 3.5.1 Docker Engine 20.10.7 Node 18 Alpine image Nuxt 3.6.5
> 
> 3.7.0 broke dependencies for me so I rolled back, but seemed to do initial build and hmr as normal.
> 
> Can you try 3.6.5?
> 
> Also can you post your docker compose file?
> 
> You can also find me on nuxt discord: peteromano

Please check my reproduction, the docker-compose file is there. Can you also share your Dockerfile and docker-compose file?

**Comment by martinszeltins** [comment]
> FYI, I exclusively use docker, it's lightning fast for me, including using nuxt layers.
> 
> Using node 18 alpine

@peteromano  Can you please share your full Docker setup (Dockerfile, docker-compose etc.)?

**Comment by or2e** [comment]
similar https://github.com/nuxt/cli/issues/118

**Comment by martinszeltins** [comment]
> similar [nuxt/cli#118](https://github.com/nuxt/cli/issues/118)

What exactly is similar to this issue there?

**Comment by martinszeltins** [comment]
**UPDATE:**
I finally found the issue. When `pnpm` installed the packages in Docker, it created a `.pnpm-store` directory in my project. This caused the watcher to go craaaazy. Once I reconfigured pnpm to place the `.pnpm-store` directory outside of my project, everything became 100% faster and working as normal.

I guess the issue was that by default in Docker pnpm puts the store directory inside the project itself. In this case Nuxt should ignore watching it just like it ignores node-modules.
